src/IO_Strings.f90

```diff
@@ -1726,4 +1726,37 @@ FUNCTION str_SDK_nDecimal(r,nDecimal) RESULT(string)
 
     ENDFUNCTION str_SDK_nDecimal
 !
+!-------------------------------------------------------------------------------
+!> @brief Returns a character array by repeating a provided sample
+!> 
+  FUNCTION charBanner(charSamp,reps) RESULT(tmpChar)
+    CHARACTER(LEN=*),INTENT(IN) :: charSamp
+    INTEGER(SIK),INTENT(IN) :: reps
+    CHARACTER(LEN=reps*LEN(TRIM(charSamp))) :: tmpChar
+    tmpChar = REPEAT(TRIM(charSamp),reps)
+  ENDFUNCTION charBanner
+!
+!-------------------------------------------------------------------------------
+!> @brief takes in a real, returns the number of digits to the left of the
+!> decimal.
+!> 
+! Primary intent of this is for writing floats properly, for that reason negative
+! numbers automatically count "-" in the returned length. However a leading zero
+! is not counted.
+  FUNCTION getFloatFormat(inputData) RESULT(lengthOut)
```

Comment:
So the goal is to basically return the number of places left of the decimal, right?  So shouldn't we really try to resolve the leading 0 too?

Reply:
Added that.